### PR TITLE
add broken geometry fixtures

### PR DESCRIPTION
**Pull request by mapsam**
This adds fixtures for #84, ported from vtzero tests in https://github.com/mapbox/vtzero/blob/088ec096dd115fa82b7e0f9cadaca26ef9c592e5/test/t/test_geometry.cpp

Includes the following fixtures:

- geometry_decoder with incomplete point 
- geometry_decoder with linestring with equal points (is this broken - unsure by the name of the fixture) 
- geometry_decoder with polygon with wrong ClosePath count 
- geometry_decoder decoding linestring with int32 overflow in x coordinate 
- geometry_decoder decoding linestring with int32 overflow in y coordinate 
- geometry_decoder with multipoint with a huge count 
- geometry_decoder with multipoint with not enough points

@joto when you have a moment, would you mind reviewing? I used vtvalidate to do most of the manual validation (thanks @GretaCB!), which is using vtzero. This confirmed whether broken geometries were fatally erroring, or being skipped.

### CODE REVIEW

File: src/049.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    layers: [
+      {
+        version: 2,
+        name: 'hello',
+        features: [
+          {
+            id: 1,
+            tags: [],
+            type: 2,
+            geometry: [ 9, 4294967294, 0, 9, 2, 2 ]
```
[the remark this answers is not in the document]
Comment (mapsam) [reply src/049.js]: Fixed

File: src/050.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    layers: [
+      {
+        version: 2,
+        name: 'hello',
+        features: [
+          {
+            id: 1,
+            tags: [],
+            type: 2,
+            geometry: [ 9, 0, 4294967295, 9, 1, 1 ]
```
[the remark this answers is not in the document]
Comment (mapsam) [reply src/050.js]: Fixed

File: src/051.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// to generate commands: http://www.cppshell.com/7gvnr
+module.exports = {
+  description: 'multipoint with a huge count value, useful for ensuring no over-allocation errors. Example error message "Max count too large"',
```
[the remark this answers is not in the document]
Comment (mapsam) [reply src/051.js]: 👍

**Comment by mapsam** [comment]
@joto appreciate the review! Good catch on those missing LineTo commands, I had written some sample scripts taken from vtzero and forgot to update the enum. 

Just to confirm, is it proper to mark the x/y overflow fixtures as `valid`? Nothing about the specification mentions maximum size, so I added a note saying they are useful for testing memory allocation in decoders.

**Comment by joto** [comment]
No, the x/y overflows have nothing to do with memory allocation. This is about overflows due to the delta encoding. This should never happen in real tiles, because the coordinates are usually between 0 and the extent, so very small numbers compared to the int32 limits, but it is undefined what should happen in those cases. vtzero just produces wrong coordinates in this case, but it would also be possible to stop processing the tile or whatever. The test should just make sure that this case is handled in some way.

**Comment by mapsam** [comment]
I see - I was confusing the large integers with the memory allocation fixtures we want in https://github.com/mapbox/mvt-fixtures/issues/73. But since this is specifically an x/y coordinate, rather than a MoveTo or LineTo command, we aren't pre-allocating anything here. Will update the descriptions!

**Comment by mapsam** [comment]
Merging! This will be part of 3.3.0

**Pull request merged**